stubs/Markdown/markdown/blockparser.pyi

```diff
@@ -1,18 +1,24 @@
-from typing import Any, List
+from typing import Any, List, TypeVar
+from xml.etree.ElementTree import Element, ElementTree
 
-class State(List[Any]):
-    def set(self, state) -> None: ...
+from . import Markdown
+from .util import Registry
+
+_T = TypeVar("_T")
+
+class State(List[_T]):
+    def set(self, state: _T) -> None: ...
     def reset(self) -> None: ...
-    def isstate(self, state): ...
+    def isstate(self, state: _T) -> bool: ...
 
 class BlockParser:
-    blockprocessors: Any
-    state: Any
-    md: Any
-    def __init__(self, md) -> None: ...
+    blockprocessors: Registry
+    state: State[Any]  # TODO: possible to get rid of Any?
+    md: Markdown
+    def __init__(self, md: Markdown) -> None: ...
     @property
-    def markdown(self): ...
-    root: Any
-    def parseDocument(self, lines): ...
-    def parseChunk(self, parent, text) -> None: ...
-    def parseBlocks(self, parent, blocks) -> None: ...
+    def markdown(self): ...  # deprecated
+    root: Element
+    def parseDocument(self, lines: list[str]) -> ElementTree: ...
+    def parseChunk(self, parent: Element, text: str) -> None: ...
+    def parseBlocks(self, parent: Element, blocks: list[str]) -> None: ...
```

Comment:
This is passed to `BlockProcessor.run()` and we could investigate which types `BlockProcessor`'s sub-classes really accept (probably `Sequence`), but I'm fine with just using `list` for now and improving later if necessary.